plugins/SitesManager/Controller.php

```diff
@@ -158,14 +159,68 @@ public function siteWithoutData()
             Piwik::postEvent('SitesManager.siteWithoutData.customizeImporterMessage', [&$googleAnalyticsImporterMessage]);
         }
 
+        list($siteType, $gtmUsed) = $this->guessSiteTypeAndGtm();
+        $instructionUrl = SitesManager::getInstructionUrlBySiteType($siteType);
+
+        $tagManagerActive = false;
+        if (Manager::getInstance()->isPluginActivated('TagManager')) {
+            $tagManagerActive = true;
+        }
+
         return $this->renderTemplateAs('siteWithoutData', array(
             'siteName'      => $this->site->getName(),
             'idSite'        => $this->idSite,
             'jsTag'         => $jsTag,
             'piwikUrl'      => $piwikUrl,
             'emailBody'     => $emailContent,
             'showMatomoLinks' => $showMatomoLinks,
+            'siteType' => $siteType,
+            'instructionUrl' => $instructionUrl,
+            'gtmUsed' => $gtmUsed,
             'googleAnalyticsImporterMessage' => $googleAnalyticsImporterMessage,
+            'tagManagerActive' => $tagManagerActive,
         ), $viewType = 'basic');
     }
+
+    private function guessSiteTypeAndGtm()
```

Comment:
This is kind of a simple method, but would be a good idea to unit test it?

Also I'm wondering if this could be done via AJAX? I'm not sure if it can exactly, but if so, it wouldn't affect page load time which might be better?

Reply:
I was thinking about the test. I'll do it.

I check the AJAX call.

Reply:
Currently, the controller renders a twig template, without any angular controller and uses some angular components (like the `piwik-content-block`). 

We can use a script tag, to make that ajax call (we use script tag checking every second if we have data), but that's not very nice, and DOM is not ready, etc, or I guess we can make a small angular component inside this plugin.

Do you want me to make it and use AJAX instead of the back-end?

Reply:
That depends on how much work you think it will be... if it's super complicated probably better not to do it for this feature since there are probably better things to work on. In that case, though, it would be good to see if there's a way to avoid having the page load time become 5+ seconds if the site is unresponsive for some reason. I guess we could cache the result of this function... if you choose to do that, take a look at the `Lazy` cache class.

(I'm not sure if there are docs about this or you already know, but there are three of these cache classes in Matomo:

- `Transient`: stored in memory for the current request only.
- `Lazy`: stored in a backend of some kind, can be file or redis or something else. The backend is loaded on request, so we only read the file or connect to redis when the specific data is requested.
- `Eager`: everything in this cache is loaded at the start of each request, so the data is readily available.
).

Reply:
@tsteur showed me `piwik-widget-loader`, I modified the PR to use that, loading the whole tabs part